### PR TITLE
753_derive_var_extreme_dt: Implement derive_var_extreme_dt(m)

### LINKED ISSUE #753: Generalize `derive_var_lstalvdt()`
Enhance `derive_var_lstalvdt()` such that the name of the new variables can be specified and the mode (first or last) can be set. It derives a datetime variable:
```
 derive_var_extreme_dtm(dataset, new_var, ... , subject_keys, mode)
```
For deriving a date variable implement `derive_var_extreme_dt()` as wrapper function calling `derive_var_extreme_dtm()` and `mutate(date())`.

### PR DESCRIPTION
**Pull request by bundfussr**
Closes #753 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

File: R/derive_var_extreme_date.R [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+  derive_var_extreme_dt(
+    dataset,
+    new_var = LSTALVDT,
+    ...,
+    source_datasets = source_datasets,
+    mode = "last",
+    subject_keys = subject_keys)
+}
+
+#' Derive First or Last Datetime from Multiple Sources
```
Comment (kabis-ops) [inline R/derive_var_extreme_date.R:45]: I think we would also want to use it to derive the first/last date from a single source?

File: R/derive_var_extreme_date.R [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(admiral.test)
+#' data("dm")
+#' data("ae")
+#' data("lb")
+#' data("adsl")
+#'
```
Comment (kabis-ops) [inline R/derive_var_extreme_date.R:113]: I would also add an example with a single source?

File: R/derive_var_extreme_date.R [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+      by_vars = subject_keys,
+      order = vars(!!new_var),
+      mode = mode,
+      check_type = "none"
+    )
+
+  left_join(dataset, all_data, by = vars2chr(subject_keys))
+}
+
+#' Derive First or Last Date from Multiple Sources
```
Comment (kabis-ops) [inline R/derive_var_extreme_date.R:281]: single source?

**Review by kabis-ops — COMMENTED** [review]
Hi Stefan, a few comments

File: R/derive_var_extreme_date.R [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(admiral.test)
+#' data("dm")
+#' data("ae")
+#' data("lb")
+#' data("adsl")
+#'
+#' # derive last known alive datetime (LSTALVDT)
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply R/derive_var_extreme_date.R]: Fixed

File: R/derive_var_extreme_date.R [hunk 5]
```diff
@@ -0,0 +1,547 @@
...
+#' dm %>%
+#'   derive_var_extreme_dt(
+#'     new_var = LSTALVDT,
+#'     ae_start, ae_end, lb_date, adsl_date,
+#'     source_datasets = list(adsl = adsl, ae = ae, lb = lb),
+#'     mode = "last"
+#'   ) %>%
+#'   select(USUBJID, LSTALVDT)
+#'
+#' # derive last alive datetime and traceability variables
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply R/derive_var_extreme_date.R]: Fixed

File: R/derive_var_extreme_date.R (hunk 1 above)
Comment (bundfussr) [reply R/derive_var_extreme_date.R:45]: Yes, you are right we discussed this. However, I think we did not consider the `derive_vars_merged*()` functions. For a single source I would rather use these. I have added them to the "See also" section.

It is not ideal that the functionality of `derive_var_extreme_dt*()` and `derive_vars_merged*()` overlaps but I have no idea how we could change it.

File: R/derive_var_extreme_date.R (hunk 3 above)
Comment (bundfussr) [reply R/derive_var_extreme_date.R:285]: See above

File: R/derive_var_extreme_date.R (hunk 1 above)
Comment (kabis-ops) [reply R/derive_var_extreme_date.R:45]: OK, let's keep this as is and see later if we can make this better.

**Review by kabis-ops — APPROVED** [review]

**Pull request merged**